ledger/src/bank_forks.rs

```diff
@@ -219,6 +234,15 @@ impl BankForks {
         );
     }
 
+    fn check_interval(root: Slot, parent_slot: Slot, snapshot_interval_slots: usize) -> bool {
```

Comment:
Just take `snapshot_interval_slots` as `Slot` so that [this](https://github.com/solana-labs/solana/pull/8314/files?utf8=%E2%9C%93&diff=unified&w=1#diff-f5031fae39eabfef864ee62e92742a1fR238) extra cast can go away?